frontend/src/pages/signup.tsx

```diff
@@ -0,0 +1,76 @@
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+
+type Props = {
+  onSuccess: () => void;
+};
+
+export default function Signup(props: Props) {
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState<Error>();
+  const router = useRouter();
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    console.log('this is the here');
+    const response = await fetch('/api/user', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ email, username, password }),
+    });
+    if (response.ok) {
```

Comment:
Maybe, it's interesting to show a custom message depending the response. Maybe we could test this.

Reply:
Added custom message for password